packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/launcher/manifest/DevLauncherManifestParser.kt

```diff
@@ -18,7 +18,14 @@ class DevLauncherManifestParser(
 
   private suspend fun downloadManifest(): Reader {
     val response = fetch(url, "GET").await(httpClient)
-    require(response.isSuccessful)
+    if (!response.isSuccessful) {
+      val extraMessage = if (url.scheme.equals("https")) {
+        "If you are trying to open a published project, you need to install a compatible version of expo-updates and follow all setup and integration steps."
+      } else {
+        "Please check your network connectivity and make sure you can access the local packager server."
+      }
+      throw Exception("Failed to open app. $extraMessage")
```

Comment:
it might be a little bit better to show both of these messages regardless of the scheme

Failed to open app.
- If you are trying to load the app from a development server, please check your network connectivity and make sure you can access the server from your device.
- If you are trying to open a publish project, ..........

Reply:
(an aside -- the term "bundler" has replaced "packager")

Reply:
use "dev server" since that's a bit more contextual here.